src/cargo/ops/cargo_update.rs

```diff
@@ -207,6 +221,215 @@ pub fn print_lockfile_changes(
+                                    dep.set_version_req(req);
+                                    return Ok(dep);
+                                } else {
+                                    trace!(
+                                        "skipping dependency `{}` without a breaking update available",
+                                        name
+                                    );
+                                }
+                                            } else {
+                                trace!(
+                                    "skipping dependency `{}` without any published versions",
+                                    name
+                                );
+                            }
+                        } else {
+                            trace!("skipping non-caret dependency `{}`: {}", name, comparator);
+                        }
+                    } else {
+                        trace!(
+                            "skipping dependency `{}` with multiple version comparators: {:?}",
+                            name,
+                            &current.comparators
+                        );
+                    }
+                } else {
+                    trace!(
+                        "skipping dependency `{}` without a simple version requirement: {}",
+                        name,
+                        version_req
+                    );
+                }
+
+                Ok(dependency)
+            },
+        )?;
+    }
+
+    Ok(upgrades)
+}
+
+/// Update manifests with upgraded versions, and write to disk. Based on cargo-edit.
+pub fn write_manifests(
+    ws: &Workspace<'_>,
+    opts: &UpdateOptions<'_>,
+    upgrades: &HashMap<String, Version>,
+) -> CargoResult<()> {
+    if !opts.breaking {
+        return Ok(());
+    }
+
+    let manifest_paths = std::iter::once(ws.root_manifest())
+        .chain(ws.members().map(|member| member.manifest_path()))
+        .collect::<Vec<_>>();
+
+    for manifest_path in manifest_paths {
+        debug!("writing manifest for `{:?}`", manifest_path);
+
+        let mut local_manifest = LocalManifest::try_new(&manifest_path)?;
+        for dep_table in local_manifest.get_dependency_tables_mut() {
+            for (dep_key, dep_item) in dep_table.iter_mut() {
+                trace!("upgrading dependency {}", dep_key);
+
+                let dep_key = dep_key.get();
+                let dependency = match crate::util::toml_mut::dependency::Dependency::from_toml(
+                    &manifest_path,
+                    dep_key,
+                    dep_item,
+                ) {
+                    Ok(dependency) => dependency,
+                    Err(err) => {
+                        opts.gctx
+                            .shell()
+                            .warn(&format!("ignoring {dep_key}, unsupported entry: {err}"))?;
+                        continue;
+                    }
```

Comment:
Unsure whether its best to error or warn

Reply:
Do you want to have it changed? It would certainly be easier to just return a CargoResult.

Reply:
You went out of your way to not fail on this.  What was the motivation?

Reply:
Good call on applying concepts from `cargo upgrade`.  However, I think the reason I did this was because it was decoupled from the development of cargo and didn't want new features in Cargo (or bugs in my code) from stopping things.

Reply:
OK, I changed it to just return an error.